### DISCUSSION TITLE
LDM Notes for December 4th, 2024

### DISCUSSION THREAD

**Discussion by 333fred** [dotnet/csharplang#8780] [LDM Notes]
https://github.com/dotnet/csharplang/blob/main/meetings/2024/LDM-2024-12-04.md

### Agenda:

- First-class span open questions
    - Preferring `ReadOnlySpan<T>` over `Span<T>` conversions
    - Conversions in expression trees

**Comment by RenderMichael** [comment]
Is there any chance of starting the “covariant array” feature down a depreciation process? The breaks won’t be silent (assignment will fail) and existing defensiveness won’t be broken (`myTArray.GetType() == typeof(T[])` will always return true; maybe we can even get an optimization which recognizes this)

**Reply by CyrusNajmabadi** [reply]
Losing variance would definitely be problematic.  Roslyn, as an example, uses it heavily. Without it, returning something that is safe and zero cost today would now incur copies.
